src/plugins/dashboard/public/locator.ts

```diff
@@ -7,14 +7,22 @@
  */
 
 import type { SerializableRecord } from '@kbn/utility-types';
+import { flow } from 'lodash';
 import type { TimeRange, Filter, Query, QueryState, RefreshInterval } from '../../data/public';
 import type { LocatorDefinition, LocatorPublic } from '../../share/public';
 import type { SavedDashboardPanel } from '../common/types';
+import type { DashboardPanelMap } from './types';
+import { convertSavedDashboardPanelToPanelState } from '../common/embeddable/embeddable_saved_object_converters';
 import { esFilters } from '../../data/public';
 import { setStateToKbnUrl } from '../../kibana_utils/public';
 import { ViewMode } from '../../embeddable/public';
 import { DashboardConstants } from './dashboard_constants';
 
+/**
+ * Useful for ensuring that we don't pass any non-serializable values to history.push (for example, functions).
+ */
+const getSerializableRecord: <O>(o: O) => O & SerializableRecord = flow(JSON.stringify, JSON.parse);
```

Comment:
shouldn't be needed as the objects we are dealing with should already be serializable.
 if not we should convert them one by on

Reply:
In a functional test I was getting a non-serializable value from `x-pack/plugins/dashboard_enhanced/public/services/drilldowns/embeddable_to_dashboard_drilldown/embeddable_to_dashboard_drilldown.tsx`.

It's a little tricky because that code does not control the full object coming in. The types indicate that `filter.meta.value` should be a string but I was getting a function in that case. Do you think we should rather move this code there or is there another way to deal with non-serializable values?

Reply:
I guess that indicates an issue with Filter (or maybe just the tests). 
When we do stringify/parse we have no idea what we are loosing from the object, so i think we should really avoid doing so.

Reply:
please open an issue to track this